lib/client-assets.php

```diff
@@ -566,7 +566,13 @@ function gutenberg_editor_scripts_and_styles( $hook ) {
 	);
 
 	// Initialize the editor.
-	wp_add_inline_script( 'wp-editor', 'wp.api.init().done( function() { wp.editor.createEditorInstance( \'editor\', window._wpGutenbergPost ); } );' );
+	$editor_settings = array(
+		'wideImages' => get_theme_support( 'wide-images' ),
+	);
+	wp_add_inline_script( 'wp-editor', 'wp.api.init().done( function() {'
+		. 'wp.editor.createEditorInstance( \'editor\', window._wpGutenbergPost, ' . json_encode( $editor_settings ) . ' ); '
```

Comment:
@youknowriad:

> But at the same time, I'd like us to avoid the globals in the Editor Code itself (to make it reusable), so a `settings` variable in `createEditorInstance` makes sense to me and could be computed using these globals.

Yes! I was thinking something similar. So perhaps each individual script could export to a global, but then we could use `jQuery.extend()` to merge them together just-in-time. For example:

```php
wp_add_inline_script( 'wp-editor', 
    sprintf(
        'wp.api.init().done( function() { wp.editor.createEditorInstance( "editor", %s ); } );',
        sprintf( 'jQuery.extend( {}, 
            { post: window._wpGutenbergPost },
            { editor: %s },
            { date: wpDateSettings },
            { i18n: wpI18nSettings },
            { themes: wpThemeSettings } )', 
            wp_json_encode( $editor_settings ) 
        )
    ) 
);
```

If `wp-editor` depends on `wp-date`, `wp-i18n`, `wp-themes`, any other scripts to which the corresponding globals are attached, then all of these globals will be guaranteed to have already been output to the browser and we'd be able to refer to them. We'd then eliminate the reliance on global variables.